src/DoubleTheDonation/API/REST/CompanyMatching.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace GiveDoubleTheDonation\DoubleTheDonation\API\REST;
+
+use Give\Donations\Models\Donation;
+use GiveDoubleTheDonation\DoubleTheDonation\FormExtension\Actions\FieldScope\HandleData;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class CompanyMatching
+{
+    public function __invoke()
+    {
+        register_rest_route(
+            'givewp/dtd',
+            'donation/(?P<id>[0-9]+)',
+            [
+                [
+                    'methods' => WP_REST_Server::EDITABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        $data = [
+                            'company_id' => $request->get_param('companyId'),
+                            'company_name' => $request->get_param('companyName'),
+                            'entered_text' => $request->get_param('enteredText')
+                        ];
+
+                        $donation = Donation::find($request->get_param('id'));
+
+                        $donationHandler = new HandleData();
+                        $donationHandler->save($data, $donation);
+                        $donationHandler->send($data, $donation);
+                    },
+                    'permission_callback' => '__return_true',
```

Comment:
Hmm, since we're interacting with the database - we should really be careful about exposing this endpoint publicly.  We should have some kind of security measure here.

Since this is intended to only be used on the confirmation page - we can try and scope it there.  

2 suggestions come to mind:
1. We can try using a nonce on the confirmation page (although, I think those are intended for being logged in)
2. We can use the `receipt-id` query param from the confirmation page which is unique to each donation.

You can then determine if a donation exists with that receipt id.

`return (bool)give()->donations->getByReceiptId($request->get_param('donation-receipt-id');`